### DISCUSSION TITLE
Wifi Primary, Secondary, return to Primary upon Restoration

### DISCUSSION THREAD

**Discussion by majorsl** [arendst/Tasmota#12319] [Q&A] [answered]
I looked through the Options and WifiConfig and can't find this, although it may already exist.

I have 2 access points configured:
-1 the primary.
-2 the secondary "backup" if the first goes down.

It seems like the current functionality will switch to 2 if 1 goes off-line, but does not switch BACK _automatically_ when 1 is detected again.

If I am correct, it would be nice if this could be added as an additional option.

**Comment by barbudor** [comment, marked as answer]
Indeed there is no fast switch back which would require Tasmota to continuously scan the SSIDs.
However you can have a look to `SetOption56` and `SetOption57` which will switch to the strongest of both AP, at restart (SO56) or during normal operations (SO57) every 44 minutes or so. You need `WifiConfig 4` for this (just like you need it also to switch from AP1 to AP2 in 1st place when AP1 fails).

Note that for Tasmota, there is no main or backup AP. AP1 and AP2 are just 2 AP without any priority. Using SO6/57 gives priority to the strongest.

**Reply by majorsl** [reply]
Ah, that's what I figured - and "priority" was the word I was looking for, thank you!

My thought was that my AP1 is the IoT wifi and AP2 is for other devices.  I don't mind a brief stay on AP2 if AP1 is down to keep functionality.  I was hoping for something along the lines of SO57 where it didn't need to be continually scanned, but a check after an interval that would put them back on the priority AP.

**Reply by sfromis** [reply]
SetOption57 is a fixed interval of 44 minutes

**Reply by sfromis** [reply]
FTR, I did a test with two AP configured, connected to first, and then turned off AP1. After some time, `SetOption57` kicked in, and the device connected to the still operational AP2. Then I turned on AP1 again (close proximity, stronger signal strength), and a bit later Tasmota came up on that again.

**Reply by majorsl** [reply]
That's great, but it assumes your preferred AP has the strongest signal which may not be in my case.  I have a preferred SSID and network configuration I want my devices attached to.  As I mentioned, short term in an outage I don't mind a fallback to AP2 then a return to AP1.... which is what I'm looking to do.  It seems the functionality is almost there - swap "strongest" for "preferred" and done.  I have no idea what code it would take to accomplish that, so I could be wishing for the impossible.

**Reply by barbudor** [reply]
@majorsl There is command to force the switching between AP (See `AP1` and `AP2`) however there is no way to force a scan to double check if AP1 is back.
If you can assume that an AP is up if it's IP address respond to ping, it should be possible to combine some rules which, once the Wifi has switched to AP2, would start pinging the IP of AP1 and force switching to AP1 is the IP responds.

**Comment by lalo-uy** [comment]
There are no Scan comman now. It is posible?
That will allow many priority  conbination in rules/script.
